src/test/java/org/cactoos/scalar/AndInThreadsTest.java

```diff
@@ -104,8 +101,10 @@ public void emptyIterator() throws Exception {
 
     @Test
     public void iteratesList() {
-        final List<String> list = Collections.synchronizedList(
-            new ArrayList<String>(2)
+        final List<String> list = new ArrayList<>(
+            new Synced<>(
+                new ArrayList<>(2)
```

Comment:
@iakunin I don't think this is going to work, you shouldn't wrap it in an `ArrayList` because it will create a complete new list. Same in the rest of this class.

Reply:
You are absolutely right. That's because of incorrect implementation of BaseEnvelope.

Now it's fixed.